File: 2225. Find Players With Zero or One Losses/Solution.py
class Solution:
    def findWinners(self, matches: List[List[int]]) -> List[List[int]]:
        loser = defaultdict(int)
        winner = defaultdict(int)
        for i in range(len(matches)):
            loser[matches[i][1]]+=1
            winner[matches[i][0]]+=1
        ls = [[],[]]
        for k,v in loser.items():
            if v ==1:
                ls[1].append(k)
        for k in winner:
            if k not in loser:
                ls[0].append(k)
        ls[0].sort()
        ls[1].sort()
        return ls

File: 2225. Find Players With Zero or One Losses/Solution.js
/**
 * @param {number[][]} matches
 * @return {number[][]}
 */
var findWinners = function(matches) {
    let wins = {};
    let lose = {}
    matches.forEach(match =>{
        wins[match[0]] = (wins[match[0]]|0) + 1;
        lose[match[1]] = (lose[match[1]]|0) + 1;
    });
    let answer = [[],[]];
    for(let m in wins){
        if (!(m in lose)){
            answer[0].push(m)
        }
    }
    for(let m in lose){
        if (lose[m] == 1){
            answer[1].push(m)
        }
    }
    answer[0].sort((a,b)=>a-b);
    answer[1].sort((a,b)=>a-b);
    return answer;
};
